server/rest/src/main/java/org/infinispan/rest/resources/ContainerResource.java

```diff
@@ -533,13 +547,31 @@ private CompletionStage<RestResponse> shutdown(RestRequest restRequest) {
    }
 
    private CompletionStage<RestResponse> listenConfig(RestRequest request) {
+      return streamConfigurationAndEvents(request, false);
+   }
+
+   private CompletionStage<RestResponse> streamListenerEvents(RestRequest request) {
+      return streamConfigurationAndEvents(request, true);
+   }
+
+   private CompletionStage<RestResponse> streamConfigurationAndEvents(RestRequest request, final boolean checkEvents) {
       MediaType mediaType = negotiateMediaType(request, APPLICATION_YAML, APPLICATION_JSON, APPLICATION_XML);
       boolean includeCurrentState = Boolean.parseBoolean(request.getParameter("includeCurrentState"));
       EmbeddedCacheManager cacheManager = invocationHelper.getRestCacheManager().getInstance();
       NettyRestResponse.Builder responseBuilder = new NettyRestResponse.Builder();
       ConfigurationListener listener = new ConfigurationListener(cacheManager, mediaType, includeCurrentState);
       responseBuilder.contentType(TEXT_EVENT_STREAM).entity(listener.getEventStream());
-      return cacheManager.addListenerAsync(listener).thenApply(v -> responseBuilder.build());
+
+      final CompletionStage<Void> cs = cacheManager.addListenerAsync(listener);
+      if (checkEvents) {
+         InternalCacheRegistry icr = registry.getComponent(InternalCacheRegistry.class);
+         Cache<?, ?> cache = icr.retrieveInternalCache(SERVER_LOGGER_CACHE_NAME);
```

Comment:
I don't think we need this new `InternalCacheRegistry` method. Instead we can do:

```java
         Cache<?, ?> cache = cacheManager.getCache(SERVER_LOGGER_CACHE_NAME);
         cs.thenCompose(v -> cache.addListenerAsync(listener));
```

With ^ I think a `null` check is redundant as the cache is always created on cache manager startup, so by the time a pod is able to serve requests the cache should always be available.

Reply:
I did that first one with the registry because I thought that the events cache was internal, but it works correctly using the cache manager directly. Should we create if it not exists? Because the `ServerEventLogger` should handle this, and if it was not created there probably means that we are using another kind of logger.